doc/ug/dv_methodology.md

```diff
@@ -299,65 +299,124 @@ Once the regression tool is developed, it will provide a way to enable these cap
   Note, an assertion precondition counts as a cover point.
-*  **Covergroup Coverage**: This metric (sometimes called "Testbench Coverage") measures covergroups during simulation.
-  These are usually coded in the testbench to check that the desired stimulus was generated properly, but can also be embedded in the RTL.
-
-Code coverage can reach 100% but it may not be sufficient to indicate whether all interesting combinations of scenarios were exercised (such as boundary conditions, concurrent scenarios and different CSR fields holding specific combinations of values resulting in the DUT being exercised in interesting ways).
-Hence, developing a detailed functional coverage model in conjunction with constrained random test sequences is key to high quality DV effort.
-
-Functional coverage can also reach 100% while the code coverage may not.
-This would indicate that the testplan is either incomplete or there are undocumented or unreachable features in the design.
-As such they can be addressed accordingly through either updating the design specification and augmenting the testplan / written tests, or by optimizing the design to remove unreachable logic if possible.
-There may be features that cannot be tested and cannot be removed from the design either.
-These would have to be analyzed and excluded from the coverage collection as a waiver.
-This is an exercise the DV and the designer typically perform together.
-This is discussed in more detail below.
+* **Cover group Coverage**:  Observes values on busses, registers, etc.
+  can verify that a specific value or a range of values was seen on a bus.
+  And that no illegal values were seen.
+  Could also be the intersection of two cover points.
+  One example could be observing the FIFO full signal and the FIFO write enable signal is asserted at the same time.
+  This is called a coverage cross and will give valuable information on whether this scenario was ever seen. 
+* **Cover Property coverage**: This observes events or series of events.
+  As an example think of the TL UL register bus used.
+  A cover property cover point could be the handshaking between valid and ready.
+
+
+### Why do we need coverage
+
+The common answer is to flush out bugs in the design.
+This is not accurate enough. Making sure there are no bugs in a design, while important it is not sufficient.
+One must also make sure the design work as intended.
+That is providing all the functionality specified in the design specification.
+So a more precise answer as to why we need coverage is to flush out flaws in the design.
+Flaws being both design bugs and deficiencies in the design with respect to the specification.
+
+#### When are we done testing?
+Another reason as to why we need coverage is to answer the seemingly simple but important question:
+**When are we done testing?**
+Do we need 1, 10, or 100 tests and should they run 10,100, or 1000 regressions?
+Only coverage can answer this question for you.
```

Comment:
This section heading is not needed. This para should be moved after explaining "Why both types of coverage is needed" below.

Reply:
I kept it following the "why do we need coverage" part as I think it better there.
but I moved the whole section to the intro.

I see the "Why both types of coverage are needed" as an elaboration of this claim.